lib/sober_swag/nodes/base.rb

```diff
@@ -13,16 +15,26 @@ class Base
 
       ##
       # Value-level comparison.
+      #
+      # @param other [Object] the other object
+      # @return [1, 0, -1] if the object is greater than, less than, or equal to the other
       def <=>(other)
-        return other.class.name <=> self.class.name unless other.class == self.class
+        return other.class.name <=> self.class.name unless other.instance_of?(self.class)
 
         deconstruct <=> other.deconstruct
       end
 
+      ##
+      # Is this object equal to the other object?
+      # @param other [Object] the object to compare
+      # @return [Boolean] yes or no
```

Comment:
[Sass' documentation](https://www.rubydoc.info/gems/sass/Sass%2FSelector%2FAbstractSequence:eql%3F) on `eql?(other)`:

```
      # Checks equality between this and another object.
      #
      # @param other [Object] The object to test equality against
      # @return [Boolean] Whether or not this is equal to `other`
```

Sass' docs also mention:
> Subclasses should define `#_eql?` rather than overriding this method,
> which handles checking class equality and hash equality.

Maybe we should take this into consideration? Then once implemented, we can add that quote to the docs as well.

Reply:
I think that's actually a SASS-only thing. I can't find any information on `#_eql?` in other ruby code bases.